File: src/main/java/org/pl/lexer/tokenizer/ComplementTokenizer.java
package org.pl.lexer.tokenizer;

import org.pl.lexer.ILexer;
import org.pl.lexer.token.ComplementToken;

public class ComplementTokenizer implements ITokenizer {
    @Override
    public boolean matches(Character c) {
        return c == '~';
    }

    @Override
    public TokenizerResult tokenize(ILexer lexer) {
        return new TokenizerResult(new ComplementToken());
    }
}

File: src/main/java/org/pl/lexer/tokenizer/ComplementTokenizer.kt
package org.pl.lexer.tokenizer

import org.pl.lexer.ILexer
import org.pl.lexer.token.ComplementToken

/**
 * Tokenizes the complement symbol '~'.
 */
class ComplementTokenizer : ITokenizer {
    override fun matches(c: Char) = c == '~'
    override fun tokenize(lexer: ILexer) = ComplementToken()
}